packages/lexical/src/LexicalNode.ts

```diff
@@ -198,6 +232,19 @@ export class LexicalNode {
   __prev: null | NodeKey;
   /** @internal */
   __next: null | NodeKey;
+  /** @internal */
+  __state: DeepImmutable<State> = {};
+
+  getState<T extends StateKey>(k: T): T['value'] {
+    const self = this.getLatest();
+    // If the state is not set, return the default value
+    return self.__state[k.key] ?? k.parse(undefined);
```

Comment:
This can be optimized later with a more complex representation of state, it's not correct as-is

```suggestion
    return k.parse(self.__state[k.key]);
```